### PR TITLE
Feature#36779 - InputNumber should support percentage

### PR DESCRIPTION
**Pull request by marcin-zapala**
ticket: https://bqmcloudrepository.visualstudio.com/App/_workitems/edit/36779

### CODE REVIEW

File: src/alto-ui/Form/InputNumber/InputNumber.js [hunk 1]
```diff
@@ -26,21 +26,34 @@ class InputNumber extends React.Component {
...
     ) {
       return {
         prev: props,
         display: state.editing
           ? state.display
-          : format(value, locale, precision, currency, disableThousandSeparator, { min, max }),
+          : format(value, locale, precision, currency, disableThousandSeparator, percent, {
+              min,
+              max,
+            }),
```
Comment (renaudtertrais) [inline src/alto-ui/Form/InputNumber/InputNumber.js]: Can you put percent in the object with min and max ? The idea is to progressively put all "options" in there

Comment (marcin-zapala) [reply src/alto-ui/Form/InputNumber/InputNumber.js]: moved to options

File: src/alto-ui/Form/InputNumber/InputNumber.js [hunk 2]
```diff
@@ -71,14 +85,20 @@ class InputNumber extends React.Component {
   }
 
   parse(value) {
-    const { locale, precision, min, max } = this.props;
+    const { locale, precision, percent, min, max } = this.props;
     const res = parse(value, locale, precision, { min, max });
-    return Number.isNaN(res) ? '' : res;
+    if (Number.isNaN(res)) return '';
+
+    return percent ? res / 100 : res;
```
Comment (renaudtertrais) [inline src/alto-ui/Form/InputNumber/InputNumber.js]: this should be done in helper/number parse() function.

Comment (marcin-zapala) [reply src/alto-ui/Form/InputNumber/InputNumber.js]: moved :)

File: src/alto-ui/helpers/number.js [hunk 3]
```diff
@@ -34,11 +34,13 @@ export const parse = (value, locale, precision = 0, { min, max } = {}) => {
...
-    x => (min || min === 0 ? Math.max(min, x) : x),
-    x => (max || max === 0 ? Math.min(max, x) : x)
-  )(num);
+  return percent
+    ? num / 100
+    : compose(
+        round(precision),
+        x => (min || min === 0 ? Math.max(min, x) : x),
+        x => (max || max === 0 ? Math.min(max, x) : x)
+      )(num);
```
Comment (renaudtertrais) [inline src/alto-ui/helpers/number.js]: you need to support min/max and precision. Best idea is to plug a new function in the compose pipe.

Comment (marcin-zapala) [reply src/alto-ui/helpers/number.js]: changed :)

**Pull request merged**